### PR TITLE
feat(blst): add new blst to all packages with bls

### PR DESCRIPTION
**Pull request by matthewkeil**
## Motivation

The @chainsafe/blst library was rebuilt with Napi to put bls work on the `libuv` thread pool in-lieu of a worker thread.  The goal is to lower the overhead for bls work and to share it across all worker threads to increase bls throughput and to lower worker thread overhead.  There will also be no latency with the messaging between workers.

## Description

Swapped out sync for async functions for all bls calls on main thread in `packages/beacon-node`

## Inclusions in `packages/beacon-node`

- update `package.json` to include `@chainsafe/blst`

## Implementation notes for review

- `getAggregatedPubkey` in `beacon-node/src/chain/bls/utils.ts` has a SWIG signatureSet (from `@lodestar/state-transition`) as param so converted in function to new binding format... not ideal but want to prevent upstream changes

These two were dependency of functions inside and outside of worker thread.  were converted to two functions (sync and async:  
- `verifySignatureSetsMaybeBatch` in `beacon-node/src/chain/bls/maybeBatch.ts`
- `getAggregatedPubkey` in `beacon-node/src/chain/bls/utils.ts`

## Files that still have `@chainsafe/bls`:

- `beacon-node/src/chain/bls/maybeBatch.ts`
- `beacon-node/src/chain/bls/utils.ts`
- `beacon-node/src/chain/bls/multithread/index.ts`
- `beacon-node/src/chain/bls/multithread/worker.ts`

### CODE REVIEW

**Comment by dapplion** [comment]
@matthewkeil Please do not replace bls everywhere, only in the hot path verifying gossip data in the beacon chain. Such that do not change any other package beyond the beacon-node

File: packages/beacon-node/package.json [hunk 1]
```diff
@@ -98,6 +98,7 @@
     "@chainsafe/as-chacha20poly1305": "^0.1.0",
     "@chainsafe/as-sha256": "^0.3.1",
     "@chainsafe/bls": "7.1.1",
+    "@chainsafe/blst": "file:../../../blst-ts-2/rebuild",
```
Comment (matthewkeil) [inline packages/beacon-node/package.json]: @dapplion I have the new bindings stubbed in using a file import.  How would you prefer I handle this for CI?  Should I publish a version to `@matthewkeil/blst` or something similar? Or perhaps an odd semVer number that we can delete from `@chainsage/blst`?

Comment (matthewkeil) [reply packages/beacon-node/package.json]: I published at `blst-ts-test` for now

**Comment by matthewkeil** [comment]
there was a build error on linux.  I am updating the bindings and I built a workflow to test all three platforms in the `@chainsafe/blst` repo.  I got my laptop setup this evening as well so it will be faster to debug the linux build issues now

**Comment by matthewkeil** [comment]
## Preliminary Results

**deployed at:** 8:30am EST
**metrics at:** 2:30pm EST (roughly.  took a few min to go through and take them all)

Not all metrics are shown here.  Just the ones that notable difference was seen (both better and worse).

### **instance:** `feat3-lg1k-hzax41`

Screen Shot 2023-05-18 at 2 16 02 PM
Screen Shot 2023-05-18 at 2 16 37 PM
Screen Shot 2023-05-18 at 2 16 50 PM
Screen Shot 2023-05-18 at 2 18 32 PM



Screen Shot 2023-05-18 at 2 16 20 PM
Screen Shot 2023-05-18 at 2 17 10 PM
Screen Shot 2023-05-18 at 2 17 21 PM
Screen Shot 2023-05-18 at 2 17 55 PM


### **instance:** `feat3-sm1v-ctvpss`

Screen Shot 2023-05-18 at 2 34 03 PM
Screen Shot 2023-05-18 at 2 34 26 PM
Screen Shot 2023-05-18 at 2 34 41 PM


Screen Shot 2023-05-18 at 2 37 27 PM
Screen Shot 2023-05-18 at 2 38 32 PM
Screen Shot 2023-05-18 at 2 38 24 PM
Screen Shot 2023-05-18 at 2 38 16 PM
Screen Shot 2023-05-18 at 2 38 08 PM
Screen Shot 2023-05-18 at 2 37 43 PM



Screen Shot 2023-05-18 at 2 38 43 PM

**Comment by dapplion** [comment]
Converting to draft as this must not be merged until extensive testing in mainnet + internal audit of the lib

**Pull request closed without merging**